extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/TransactionManagerConfiguration.java

```diff
@@ -32,6 +32,12 @@ public final class TransactionManagerConfiguration {
     @ConfigItem(defaultValue = "ObjectStore")
     public String objectStoreDirectory;
 
+    /**
+     * Enable recovery service to start on startup
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean enableXaRecovery;
```

Comment:
Do we have to restrict recovery to just JTA? If we call it `enableRecovery` then the same property will work for all of our transaction models (in fact, enabling recovery for just JTA would add some complexity).

BTW I argued on the other PR (https://github.com/quarkusio/quarkus/pull/28003#issuecomment-1252185392) that making recovery configurable is dangerous and potentially exposes JTA users to non-ACID behaviour.

Reply:
We can rename it to `enableRecovery` if you prefer. My understanding is that it only made sense for XA. But if not, I can rename it. Just let me know soon because the window is closing fast.

The situation is quite simple: this patch had annoying consequences for our users, so it was either revert it altogether for 2.13 and find solutions in the future or find a compromise for 2.13 and keep the patch in there (which doesn't prevent more discussions and improvements in the future). We went with the latter.

Reply:
I'm renaming the property and will push soon. I think we need a discussion with you, @zhfeng , @Sanne to decide what we want to do for 2.14 with the recovery service, when it is needed, when it should be enforced by default...